src/Mollie/WC/Gateway/Abstract.php

```diff
@@ -2136,4 +2141,104 @@ protected function appendOrderArgumentsToUrl($order_id, $order_key, $webhook_url
         );
         return $webhook_url;
 }
+
+    /**
+     * @param $new_order_status
+     *
+     * @param $statusTypeString
+     *
+     * @return mixed|void|null
+     */
+    protected function overwriteFilters($new_order_status, $statusTypeString)
+    {
+        // Overwrite plugin-wide
+        $new_order_status = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString,
+            $new_order_status
+        );
+
+        // Overwrite gateway-wide
+        $new_order_status = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString . $this->id,
+            $new_order_status
+        );
+        return $new_order_status;
+    }
+
+    /**
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     *
+     * @return bool
+     */
+    protected function isPartialRefund($payment)
+    {
+        return ($payment->amount->value - $payment->amountRefunded->value) != 0;
+    }
+
+    /**
+     * @param WC_Order                                                $order
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     */
+    protected function processUpdateStateRefund(WC_Order $order, $payment)
+    {
+        if ($this->isPartialRefund($payment)) {
+            $this->updateStateRefund(
+                $order,
+                $payment,
+                self::STATUS_ON_HOLD,
+                '_order_status_partially_refunded',
+                'partially refunded'
+            );
+            return;
+        }
+        $this->updateStateRefund(
+            $order,
+            $payment,
+            self::STATUS_REFUNDED,
+            '_order_status_refunded',
+            'refunded'
+        );
```

Comment:
```suggestion
$this->isPartialRefund($payment) 
    ? $this->updateStateRefund(
        $order,
        $payment,
        self::STATUS_ON_HOLD,
        '_order_status_partially_refunded',
        'partially refunded'
    )
    : $this->updateStateRefund(
        $order,
        $payment,
        self::STATUS_REFUNDED,
        '_order_status_refunded',
        'refunded'
    );
```

Reply:
Also, I think the `partially refunded` and `refunded` must be a translated string.